### DISCUSSION TITLE
Is RTSP-Simple-Server supports RTSP Over HTTPS both at client and server side?

### DISCUSSION THREAD

**Discussion by Vukia** [bluenviron/mediamtx#1337] [Questions]
Hi all,
I have to build a RTSP proxy server to collect ip camera streams and ensure all packets are encrypted in transporting.
I know camera supported RTSPS protocol is the simplest and best solution for this requirement, but there is just a few 
brands and cameras that support RTSPS.
I've tested VLC but VLC seems just support Over-HTTP, HTTPS is not worked.
Thanks for any information.

**Comment by aler9** [comment]
Hello, you're perfectly right about VLC limitations, but at the same time i don't want to start digging into RTSP via HTTPS, because there are no certainties about VLC support or support by any other software.

If you don't want to allocate a RTSPS -> RTSP converter on the same machine on which VLC is running, as suggested in the README, in my opinion you can enable HLS encryption and read the HLS stream with VLC, which supports reading encrypted HLS streams.

Server configuration:

```yml
hlsEncryption: yes
hlsServerKey: server.key
hlsServerCert: server.crt
```

URL to read with VLC:

```
https://rtspss:8888/mystream
```

In this way communication is encrypted.

[a comment by an automated account is not shown]
